framework/infrastructure/gcp/cloud_run.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2025 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+import logging
+
+from googleapiclient import discovery
+
+from framework.infrastructure import gcp
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TEST_PORT = 8080
+DEFAULT_TIMEOUT = 600
+
+
+class CloudRunApiManager(
+    gcp.api.GcpStandardCloudApiResource, metaclass=abc.ABCMeta
+):
+    project: str
+    region: str
+    _parent: str
+    service: discovery.Resource
+    api_manager: gcp.api.GcpApiManager
+
+    def __init__(self, project: str, region: str):
+        if not project:
+            raise ValueError("Project ID cannot be empty or None.")
+        if not region:
+            raise ValueError("Region cannot be empty or None.")
+        self.api_manager = gcp.api.GcpApiManager(
+            v2_discovery_uri="https://run.googleapis.com/$discovery/rest?"
+        )
+        self.project = project
+        self.region = region
+        service: discovery.Resource = self.api_manager.cloudrun("v2")
+        self.service = service
+        self._parent = f"projects/{self.project}/locations/{self.region}"
+        super().__init__(self.service, project)
+
+    @property
+    def api_name(self) -> str:
+        """Returns the API name for Cloud Run."""
+        return "run"
+
+    @property
+    def api_version(self) -> str:
+        """Returns the API version for Cloud Run."""
+        return "v2"
+
+    def create_cloud_run_resource(
+        self, service: discovery.Resource, service_name: str, body: dict
+    ):
+        service.projects().locations().services().create(
+            parent=self._parent, serviceId=service_name, body=body
+        ).execute()
+
+    def get_cloud_run_resource(
+        self, service: discovery.Resource, service_name: str
+    ):
```

Comment:
Instead, return a dataclass object. See `framework.infrastructure.gcp.network_services.NetworkServicesV1.get_mesh` and `framework.infrastructure.gcp.network_services.Mesh` for example.

Feel free to only parse fields that you need.